ocaml/xenopsd/lib/xenops_sandbox.mli

```diff
@@ -0,0 +1,52 @@
+module Chroot : sig
+  type t = private {root: string; uid: int; gid: int}
+
+  module Path : sig
+    type t
+
+    val of_string : relative:string -> t
+  end
+
+  val absolute_path_outside : t -> Path.t -> string
+  (** [absolute_path_outside chroot path] returns the absolute path outside the
+      chroot *)
+
+  val chroot_path_inside : Path.t -> string
+  (** [chroot_path_inside path] returns the path when inside the chroot *)
+
+  val of_domid : daemon:string -> domid:int -> vm_uuid:string -> t
```

Comment:
The is indexed on both domid and uuid; I believe services are per domain, so domid would be enough. What creates the need to have the uuid as well? (Obviously we could also look up the domid for a uuid but not involving the database is a good idea)

Reply:
the root directory needs both, unfortunately:
```ocaml
    let root =
      if domid = 0 then
        Printf.sprintf "/var/run/xen/%s-root-%d-%s" daemon domid vm_uuid
      else
        Printf.sprintf "/var/run/xen/%s-root-%d" daemon domid
```

Reply:
I think that is just to guard against domid reuse (e.g. if your domid has overflown because you've exceed the max domid - which is only 16-bits), to avoid accidentally using another VM's chroot (that may still have services that didn't get cleaned up). Although if we're careful about cleaning up chroot as domains die we don't need it, I thought it'd be worth having a bit of an extra layer of defense to avoid an XSA in the future.